simexpal/queuesock.py

```diff
@@ -56,20 +59,41 @@ def run(self):
 							self._should_stop = True
 						elif request['action'] == 'kill':
 							print("Terminating current subprocess")
-							if cur_subprocess is not None:
-								cur_subprocess.terminate()
+							if cur_subproc is not None:
+								cur_subproc.terminate()
 							self.close()
 							return
+						elif request['action'] == 'show':
+							pending_runs = []
+							for request in requests:
+								specfile_path = request['specfile_path']
+								with open(specfile_path, 'r') as f:
+									manifest = util.read_yaml_file(f)['manifest']
+
+								pending_runs.append(self.get_run_display_name(manifest))
+
+							queue_info = {
+								'current_run': cur_run,
+								'pending_runs': pending_runs,
+								'num_completed_runs': num_completed_runs
+							}
+							sk.data.send(util.yaml_to_string(queue_info))
 						else:
 							print("Ignoring request with unknown action '{}': {}".format(request['action'], request))
 
 					sk.data.close()
 					self.selector.unregister(sk.fd)
 
-			if cur_subprocess is not None:
-				subprocess_terminated = cur_subprocess.poll() is not None
+			if cur_subproc is not None:
+				cur_subproc_terminated = cur_subproc.poll() is not None
+
+			if cur_subproc_terminated:
+
+				if cur_subproc != subproc:
+					num_completed_runs += 1
+					subproc = cur_subproc
+					cur_run = None
```

Comment:
Huh? What does this do? Comparing subprocesses looks strange.

Reply:
It was supposed to make sure that the number of completed runs is counted correctly. I thought about it again and was able to omit the ``subproc`` variable. Simply comparing ``cur_subproc`` to ``None`` and resetting ``cur_subproc`` is enough.